File: Java/src/main/java/com/gildedrose/OtherItem.java
package com.gildedrose;

public class OtherItem extends ItemStrategy {

    @Override
    public void update(Item item) {
        if (item.quality > 0) {
            decrementQuality(item);
        }
        decrementSellin(item);
        if (item.sellIn < 0) {
            if (item.quality > 0) {
                decrementQuality(item);
            }
        }
    }
}

File: csharp/otherItem.cs
namespace GildedRose
{
    public class OtherItem : ItemStrategy
    {
        public override void Update(Item item)
        {
            DecrementQuality(item);

            DecrementSellin(item);

            if (item.SellIn < 0 )
            {
                DecrementQuality(item);
            }
        }
    }
}